addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -88,3 +89,39 @@ def test_notify_reply_to_computation_mc(self):
                     "%s %s" % (self.user_employee_c2.company_id.name, test_record.name),
                     "%s@%s" % (self.alias_catchall, self.alias_domain)))
             )
+
+@tagged('multi_company')
+class TestMultiCompanySetupAttachment(TestMailCommon, TestRecipients):
+    """ Test message post with attachment on record of another company than the
+    user company but for which the user has readonly access on. """
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._activate_multi_company()
+
+        cls.test_record = cls.env['mail.test.simple.mc'].with_context(cls._test_context).create({
+            'company_id': cls.user_employee.company_id.id,
+            'email_from': 'ignasse@example.com',
+            'name': 'Test',
+        }).with_context({})
+
+    @users('employee_c2')
+    def test_post_main_attachment_on_record_with_read_access(self):
+        """ Check that with readonly access, a message with attachment can be
+        posted on a model with the attribute _mail_post_access = 'read'. """
+        test_record = self.env['mail.test.simple.mc'].browse([self.test_record.id])
+        test_record.message_post(attachments=[('testAttachment', b'Test attachment')])
+
+    @users('employee_c2')
+    def test_post_main_existing_attachment_on_record_with_read_access(self):
+        """ Check that with readonly access, a message with pre-created attachment
+        can be posted on a model with the attribute _mail_post_access = 'read'. """
+        test_record = self.env['mail.test.simple.mc'].browse([self.test_record.id])
+        test_record.message_post(attachment_ids=[self.env['ir.attachment'].create({
```

Comment:
Can't we merge it with the other test, and have a single test with both attachment_ids and attachments tuple in it ? That way we cover both use cases in a single "template-like" test (like when using a template with an attachment and a report that generates both of them).